lib/collection.js

```diff
@@ -0,0 +1,290 @@
+        this.set(data);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    isChanged: function (branch) {
+        return this._models.some(function (model) {
+            return model.isChanged(branch);
+        });
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Boolean}
+     */
+    commit: function (branch) {
+        return this._models.reduce(function (changed,  model) {
+            return model.commit(branch) || changed;
+        }, false);
+    },
+
+    /**
+     * @param {String} [branch=DEFAULT_BRANCH]
+     * @returns {Collection}
+     */
+    revert: function (branch) {
+        this._models.forEach(function (model) {
+            model.revert(branch);
+        });
+        return this;
+    },
+
+    /**
+     * @returns {Array.<Object>}
+     */
+    toJSON: function () {
+        return this._models.map(function (model) {
+            return model.toJSON();
+        });
+    },
+
+    /**
+     * @param {Number} index
+     * @returns {Model}
+     */
+    at: function (index) {
+        return this._models[index];
+    },
+
+    /**
+     * @param {*} id
+     * @returns {?Model}
+     */
+    get: function (id) {
+        return this._modelsIdsMap[id] || null;
+    },
+
+    /**
+     * @param {Function} fn
+     * @param {Object} [ctx]
+     * @returns {?Model}
+     */
+    find: function (fn, ctx) {
+        var found = null;
+        this.some(function (model, index, array) {
+            var isFound;
+            if (ctx) {
+                isFound = fn.call(ctx, model, index, array);
+            } else {
+                isFound = fn(model, index, array);
+            }
+            if (isFound) {
+                found = model;
+                return true;
+            }
+        });
+        return found;
+    },
+
+    /**
+     * @param {Object} conditions
+     * @returns {Array.<Model>}
+     */
+    where: function (conditions) {
+        return this.filter(this._isMatchConditions.bind(this, conditions));
+    },
+
+    /**
+     * @param {Object} conditions
+     * @returns {?Model}
+     */
+    findWhere: function (conditions) {
+        return this.find(this._isMatchConditions.bind(this, conditions));
+    },
+
+    /**
+     * @param {String} attr
+     * @returns {Array}
+     */
+    pluck: function (attr) {
+        return this.map(function (model) {
+            return model.get(attr);
+        });
+    },
+
+    /**
+     * @param {(Model|Object|Array.<(Model|Object)>)} models
+     * @param {Object} [options]
+     * @param {Number} [options.at]
+     * @returns {Collection}
+     */
+    add: function (models, options) {
+        var at = options && options.at;
+        models = Array.isArray(models) ? models : [models];
+        models.forEach(function (model, index) {
+            var currentOptions;
+            model = this._prepareModel(model);
+            if (!this._isExists(model)) {
```

Comment:
> 1) what should return get(id) method in this case?
Really don't know.  It happens so, that get(id) feature create implicit restriction that model ids should be unique. 
We can
1) make this a feature describing this behaviour in docs
2) implement method get  as it returns first found model with such id and method getAll(id), that returns all models with such id.